modules/authentication/proc-enabling-authentication-with-github.adoc

```diff
@@ -0,0 +1,149 @@
+Prefer a GitHub App to an OAuth app to use fine-grained permissions, give more control over which repositories the application can access, and use short-lived tokens.
+
+.. link:https://docs.github.com/en/apps/creating-github-apps/registering-a-github-app/registering-a-github-app[Register a GitHub App] with the configuration:
++
+* *GitHub App name*: Enter a unique name identifying your GitHub App, such as __<{product}>__-__<GUID>__.
+* *Homepage URL*: Your {product-short} URL: `pass:c,a,q[{my-product-url}]`.
+* *Authorization callback URL*: Your {product-short} authentication backend URL: `pass:c,a,q[{my-product-url}/api/auth/github/handler/frame]`.
+* *Webhook URL*: Your {product-short} URL: `pass:c,a,q[{my-product-url}]`.
+* *Webhook secret*: Provide a strong secret.
+* *Repository permissions*:
+** Enable `Read-only` access to:
+*** *Administration*
+*** *Commit statuses*
+*** *Contents*
+*** *Dependabot alerts*
+*** *Deployments*
+*** *Pull Requests*
+*** *Webhooks*
++
+TIP: When you intend to make changes via the GitHub API, enable `Read and write` permissions rather than `Read-only`.
+
+** Toggle other permissions as per your needs.
+
+* *Organization permissions*:
+** Enable `Read-only` access to:
+*** *Members*
+
+* *Where can this GitHub App be installed?*: Select `Only on this account`.
+
+.. In the *General* tab, *Clients secrets* section, click *Generate a new client secret*.
+
+.. In the *General* tab, *Private keys* section, click *Generate a private key*.
+
+.. In the *Install App* tab, choose an account to install your GitHub App on.
+
+.. Save for the next step:
+
+* **App ID**
+* **Client ID**
+* **Client secret**
+* **Private key**
+* **Webhook secret**
+
+. To add your GitHub credentials to your {product-short} secrets, edit your {product-short} secrets, such as `secrets-rhdh`, and add the following key/value pairs:
++
+`AUTH_GITHUB_APP_ID`:: Enter the saved **App ID**.
+`AUTH_GITHUB_CLIENT_ID`:: Enter the saved **Client ID**.
++
+Optional: enter additional secrets, not required for authentication, but required for further integration with GitHub:
++
+`GITHUB_HOST_DOMAIN`:: Enter your GitHub host domain: `pass:c[https://github.com]` unless you are using GitHub Enterprise.
+`GITHUB_ORGANIZATION`:: Enter your GitHub organization name, such as `__<your_github_organization_name>__'.
+`GITHUB_ORG_URL`:: Enter `$GITHUB_HOST_DOMAIN/$GITHUB_ORGANIZATION`.
+`GITHUB_CLIENT_SECRET`:: Enter the saved **Client Secret**.
+`GITHUB_PRIVATE_KEY_FILE`:: Enter the saved **Private key**.
+`GITHUB_WEBHOOK_SECRET`:: Enter the saved *Webhook secret*.
+
+. To set up the GitHub authentication provider in your {product-short} custom configuration, edit your custom {product-short} ConfigMap such as `app-config-rhdh`, and add the following lines to the `app-config-rhdh.yaml` content:
++
+--
+.`app-config-rhdh.yaml` fragment with mandatory fields to enable authentication with GitHub
+[source,yaml]
+----
+auth:
+  environment: production
```

Comment:
@themr0c Consider using number tags to the code snippets and descriptions.

Reply:
Intentionally, no, to avoid garbage when a user copies the YAML.